Redefinition of __rdtsc breaks modern GCC

Platform: Ubuntu 22.04 LTS
Compiler: gcc (Ubuntu 11.2.0-19ubuntu1) 11.2.0

gcc 11 now defines __rdtsc which results in a compile error.
```
src/utils.h:60:34: error: ‘long long unsigned int __rdtsc()’ redeclared inline without ‘gnu_inline’ attribute
   60 | static inline unsigned long long __rdtsc(void)

```

--- a/src/utils.h
+++ b/src/utils.h
@@ -48,7 +48,7 @@
 
 #define xassert(cond, msg) if ((cond) == 0) _err_fatal_simple_core(__func__, msg)
 
-#if defined(__GNUC__) && !defined(__clang__)
+#if defined(__GNUC__) && __GNUC__ < 11 && !defined(__clang__)
 #if defined(__i386__)
 static inline unsigned long long __rdtsc(void)
 {

